packages/scripts/utils/env.js

```diff
@@ -176,7 +176,7 @@ function buildWordPress( newInstall, fastInstall ) {
 
 		if ( env.npm_package_wp_env_welcome_build_command ) {
 			const nextStep = sprintf(
-				'\nRun %s to build the latest version of %s, then open %s to get started!\n',
+				'\nRun %1$s to build the latest version of %2$s, then open %3$s to get started!\n',
```

Comment:
This is interesting 🤔 It kinda seems like it was perfectly valid before these changes, no?

Raising the questions:

- Should it be exclusive to use within translation functions?
- Should it be exclusive to applying to `sprintf` from `wp-i18n` specifically (where we can better assume it's relevant for use in a translation string), not if imported via `sprintf-js`.

Reply:
Hmm good question. I‘d say it‘s technically valid, although numbering/naming would make it easier to tweak (best practice). For translated strings it would be necessary.

Not sure whether we should make that assumption...

Reply:
I don't know that we need to do it within the scope of this pull request, but we should enhance the rule to consider `sprintf` only from `@wordpress/i18n`, as a "solution" to allow the original usage here.

Reply:
~Do you perhaps have some starting points for me on how to achieve this?
Detecting `i18n.sprintf()` seems easy. But if it's `import { sprintf } from '@wordpress/i18n'` or `const { sprintf } = wp.i18n` it looks like I need to traverse through the AST? The import statement would be at the top lavel, but the latter could occur anywhere, no?~

Thinking about it more, isn't it unlikely for a regular project to use both `sprintf-js` and `sprintf` from `@wordpress/i18n`? The docs for the `valid-sprintf` rule explicitly link to the `@wordpress/i18n` package. So if you use that rule it's fair to assume `sprintf` comes from that package and it's up to the developer to either a) disable the rule if not applicable or b) fix the issues it legitimately reports.

Reply:
> Thinking about it more, isn't it unlikely for a regular project to use both `sprintf-js` and `sprintf` from `@wordpress/i18n`? The docs for the `valid-sprintf` rule explicitly link to the `@wordpress/i18n` package. So if you use that rule it's fair to assume `sprintf` comes from that package and it's up to the developer to either a) disable the rule if not applicable or b) fix the issues it legitimately reports.

My original thinking with raising the example here is because we _are_ explicitly using `sprintf-js` in this instance.

https://github.com/WordPress/gutenberg/blob/5941c924425f1d09bc724652cc316f5df90d4d32/packages/scripts/utils/env.js#L8

Reply:
Right. Then we should IMO either disable the `valid-sprintf` rule for that package (as it's not _that_ `sprintf`) or, because it's still `sprintf`, fix the issues as good practice to improve code readability & maintainability.

Not worth the hassle trying to detect only `sprintf` coming from  `@wordpress/i18n`

Reply:
Fair enough. It does feel like using the name of the function is a pretty loose association. For example, I assume these rules could trigger if I decide to implement and use my own custom `sprintf` function that has nothing to do with `@wordpress/i18n`. As you note, this is a pretty rare case, and likely not something worth spending too much effort on.